### DISCUSSION TITLE
Passing scattered S3 objects into a ContainerTask

### DISCUSSION THREAD

**Discussion by djbutler** [flyteorg/flyte#4100] [Q&A] [answered]
I have a list of S3 objects that I need to pass into a ContainerTask, and ideally I'd like Flyte to handle the downloading rather than having the user code handle that. If all the S3 objects were in the same directory (same prefix), I could handle this with a simple FlyteDirectory. But these S3 objects are potentially scattered across different directories. Is there a construct like a List[FlyteFile] that would handle this use case?

**Comment by Future-Outlier** [comment, marked as answer]
this is fixed, now to support FlyteDirectory as input output in flyte
